api/grpcserver/grpcserver_test.go

```diff
@@ -1628,28 +1643,30 @@ func TestTransactionServiceSubmitUnsync(t *testing.T) {
 	shutDown := launchServer(t, grpcService)
 	defer shutDown()
 
-	conn := dialGrpc(t, cfg)
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	conn, err := grpc.DialContext(ctx,
+		"localhost:"+strconv.Itoa(cfg.GrpcServerPort),
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+		grpc.WithBlock(),
+	)
+	require.NoError(t, err)
+	t.Cleanup(func() { require.NoError(t, conn.Close()) })
```

Comment:
IMO, grpc connection should have a separate timeout. 10s for connecting makes no sense.
```suggestion
	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
	defer cancel()
	conn, err := grpc.DialContext(ctx,
		"localhost:"+strconv.Itoa(cfg.GrpcServerPort),
		grpc.WithTransportCredentials(insecure.NewCredentials()),
		grpc.WithBlock(),
		grpc.WithTimeout(time.Second),
	)
	require.NoError(t, err)
	t.Cleanup(func() { require.NoError(t, conn.Close()) })
```

Reply:
|grpc.WithTimeout` is functionally the same as `context.WithTimeout`. The functional option is marked as deprecated and reported by staticcheck if used.

Reply:
> 10s for connecting makes no sense.

Why not? The timeout isn't only for `grpc.DialContext` but for the whole test. If connecting takes 5 seconds and the remaining test finishes in under 5 seconds is that not just as valid as if connecting takes 1 second?

Reply:
Because connecting shouldn't take 10s. I'd use a separate ctx for the test with timeout 10s and a child ctx for connection with timeout 1s.